Add line rendering plus name labels for ridges and aretes

jeisenbe opened the pull request:
Related to #545 and #788
Also see discussion in #1148 and #2774 
Replaces part of PR #2138

# Changes proposed in this pull request:
- Add a pattern-based linear rendering for ridges and a similar rendering for aretes
- Render text labels for named ridges and aretes in the same style used for cliffs and embankments

## Explanation
- Ridges and aretes are well-defined linear landform features
- Both are approved tags with complete wiki pages and increasing usage by mappers in many areas
- Other renderings currently render text labels (eg [opentopomap](https://github.com/der-stefan/OpenTopoMap/blob/master/mapnik/styles-otm/text-natural-line.xml)) or lines ([Osmand](https://github.com/osmandapp/OsmAnd-resources/pull/470)) for ridges or aretes
- In some cases mappers have used `natural=cliff` to map ridges and aretes, perhaps influenced by the lack of rendering of `natural=ridge` and `natural=arete` in this style. Some named `natural=peak` and `natural=mountain_range` could be better mapped as ridges, in the case that the name refers to one linear hill or mountain feature, but the lack of rendering may discourage this.
- Adding rendering of these features in this style will promote correct mapping by showing the geometries in the database and the name labels

However, a pleasant and clear rendering is not easy. A previous PR was closed due to issues about the initial zoom levels, name label size, and the appearance of the ridges.

This new attempt is initially based off of work by @imagico in the [osm-carto-alternative-colors style](github.com/imagico/osm-carto-alternative-colors/). 

# Options with test renderings
## 1) [osm-carto-alternative-colors style](github.com/imagico/osm-carto-alternative-colors/)
- Aretes and ridges are rendered from z13 with 2 patterns. Aretes have a wider pattern at z15 and above (similar to how cliff lines change size at this point), but the ridge pattern is the same.
- I've also added text labels for the name at z15 and above, in the same style used for cliffs and embankments (gray color).
- Uses SQL to slightly shorten the ways when they intersect saddles and peaks. This makes it easier to see the peak and saddle icons, but can be confusing at times. 
- Also, it appears that some ridges are not rendered when they are also tagged as a boundary=administrative - I'm not sure if this is a problem with the code I borrowed from @imagico or a mistake in my implementation

### z14 Mt Everest aretes - alt-colors style - thin
![z14-everest-alt-colors-aretes](https://user-images.githubusercontent.com/42757252/56782578-b7014000-6822-11e9-8839-77f4503b7a31.png)
### z15 Mt Everest aretes - alt-colors style - wide
![z15-everest-alt-colors](https://user-images.githubusercontent.com/42757252/56782594-c2ed0200-6822-11e9-8ecc-44c1beae5419.png)

### z13 Parchemuche, Nepal ridges - alt-colors style
![z13-parchemuche-alt-colors](https://user-images.githubusercontent.com/42757252/56782604-cc766a00-6822-11e9-8318-0d1fd04bf7b6.png)
### z13 Kongde Ri, Nepal ridges- alt-colors style
![z13-kongde-ri-alt-colors](https://user-images.githubusercontent.com/42757252/56782606-ce402d80-6822-11e9-864b-c37ce8bbb53a.png)


## 2) Modified alt-colors style
- Here I've simplified the SQL query. The aretes and ridges now are rendered at full length.
- While it's a little harder to see saddles and peaks along a ridge, this problem also exists for peaks rendered over administrative boundaries, paths, etc, and could be better addressed by adding a halo around the saddle and peak icons or changing the icons to a darker, simpler design.
- The patterns are the same as in 1): Aretes and ridges are rendered from z13. Aretes have a wider pattern at z15. Name labels are rendered from z15
- I'm still not sure about this arete pattern: it's directional, and reverses depending on how the osm way is drawn.

### z14 Mt Everest aretes - no gap - thin
![z14-everest-aretes-no-gap](https://user-images.githubusercontent.com/42757252/56782722-4eff2980-6823-11e9-9bf5-ac90a57fae92.png)
### z15 Mt Everest aretes - no gap - wide
![z15-everest-aretes-no-gap](https://user-images.githubusercontent.com/42757252/56782727-51fa1a00-6823-11e9-9342-d42280c0773b.png)

### z13 Parchemuche, Nepal ridges - no gap
![z13-parchemuche-no-gap](https://user-images.githubusercontent.com/42757252/56782726-51fa1a00-6823-11e9-8fde-3118aab49573.png)
### z13 Kongde Ri, Nepal ridges - no gap
![z13-kongde-ri-no-gap](https://user-images.githubusercontent.com/42757252/56782723-50305680-6823-11e9-9094-4005651eb7bb.png)
### z15 - no gap
![z15-ridges-borders-glacier-no-gap](https://user-images.githubusercontent.com/42757252/56782732-59212800-6823-11e9-9a55-b8d0603cff79.png)


## 3) Narrower ridges at z13 and z14 and staggered arete pattern
- Where I live, near the equator, the 28 pixel wide ridge pattern is ~400m wide at z13. I've narrowed the ridge pattern to 18 pixels wide and used this for z13 and z14
- The original arete pattern was directional, so I've tried a different option with staggered v shapes on each side. This was created by moving around the "rocks" in the original ridge svg
- However, the staggered v-shaped arete line can give an optical illusion of the ridge line zig-zagging back and forth

### z13 Mt Everest aretes - alternating pattern - medium width
![z13-everest-arete-alternating-mid](https://user-images.githubusercontent.com/42757252/56782849-e795a980-6823-11e9-920c-b627ded95253.png)
### z14 Mt Everest aretes - alternating pattern - medium width
![z14-everest-arete-mid-alternating](https://user-images.githubusercontent.com/42757252/56782853-ea909a00-6823-11e9-9bd2-d4caade1e1f1.png)
### z15 Mt Everest aretes - alternating pattern - wide
![z15-everest-aretes](https://user-images.githubusercontent.com/42757252/56782872-fc723d00-6823-11e9-817b-167855645f9d.png)

### z13 Parchemuche, Nepal ridges - medium width
![z13-ridges-parchemuche-mid](https://user-images.githubusercontent.com/42757252/56782967-64288800-6824-11e9-944e-2f5545567026.png)

### z13 Kongde Ri, Nepal ridges - medium width
![z13-ridges-mid-kongde-ri](https://user-images.githubusercontent.com/42757252/56782974-6a1e6900-6824-11e9-9de8-97e2c3d20606.png)

### z15 - wide (same as 1) and 2) above)
![z15-ridges-borders-glacier-no-gap](https://user-images.githubusercontent.com/42757252/56782732-59212800-6823-11e9-9a55-b8d0603cff79.png)

## 4) Even thinner ridges and aretes at z13
- This option tried thinner ridges and aretes z13, and the medium thickness at z14, requiring 3 different pattern files. 
- However, I think this may be too light and hard to see

### z13 Mt Everest aretes - thinnest
![z13-everest-aretes-thinnest](https://user-images.githubusercontent.com/42757252/56783068-d00af080-6824-11e9-9d0c-0975cb51f3d5.png)

### z14 mid-width aretes
![z14-everest-arete-mid-alternating](https://user-images.githubusercontent.com/42757252/56782853-ea909a00-6823-11e9-9bd2-d4caade1e1f1.png)

### z13 Konge Ri ridges - thinnest
![z13-everest-aretes-thinnest](https://user-images.githubusercontent.com/42757252/56783114-07799d00-6825-11e9-87d9-07562935966a.png)

### z14 mid-width ridges
![z14-ridges-thin-kongde-ri](https://user-images.githubusercontent.com/42757252/56783168-40197680-6825-11e9-8f43-e74d12618fb9.png)


## 5) Symmetrical arete pattern
- As mentioned, the staggered or alternating arete pattern causes a bit of an optical illusion that causes the line to appear to zig-zag.
- Here the aretes are rendered with the "V" shapes directly opposite each other, so that the ridge line still appears straight
- However, it looks like I still need to improve the SVG to look more even and so that there is not a cut-off appearance on one end of the arete
- 2 ridge patterns: `ridge-mz.svg` for z13 and z14, `ridge-hz.svg` for z15 and above; 2 arete patterns.

### z13 Mt Everest aretes - symmetrical medium-thin pattern
![z13-everest-symmetrical-aretes](https://user-images.githubusercontent.com/42757252/56783207-7820b980-6825-11e9-88ab-b361f6df0027.png)

### z15 Mt Everest aretes - symmetrical wider pattern
![z15-everest-aretes-symmetrical](https://user-images.githubusercontent.com/42757252/56783210-7b1baa00-6825-11e9-8f7a-9602a07f7b81.png)

### z16 Northeast Ridge, Everest - need to adjust the pattern so it doesn't look cut-off (right side)
![z16-ne-ridge-arete-symmetrical](https://user-images.githubusercontent.com/42757252/56783227-a2727700-6825-11e9-8a3d-0af18cc659d7.png)


## Test areas
- Mt Everest aretes: https://www.openstreetmap.org/#map=13/27.9864/86.9292
- Ridges in the central Himalaya: https://www.openstreetmap.org/#map=13/27.8088/86.5408
- Tom Ukul in Papua, Indonesia: https://www.openstreetmap.org/#map=13/-4.2231/138.9199
- Pasir Putih, Papua, Indonesia: https://www.openstreetmap.org/#map=14/-4.04133/138.9413
- Wosi, Papua, Indonesia: https://www.openstreetmap.org/#map=13/-3.9272/138.8693
- Jimliek, Papua, Indonesia: https://www.openstreetmap.org/#map=13/-3.8028/138.8256

## Remaining questions
- What arete pattern should be used?
- Should ridges be rendered with a thinner pattern at z13 and z14?

jeisenbe commented:
## 6) Improved symmetrical arete pattern
- New SVGs `arete-wide.svg` for >=z15 and `arete-mid.svg` for z13-z14
- This commit uses the same ridge patterns as 3) and 5) above: narrow at z13-z14, wide at >=z15

**z13 Everest aretes** - improved symmetrical pattern - mid width
![z13-everest-aretes-smoother-symmetrical](https://user-images.githubusercontent.com/42757252/56787743-6b599100-6838-11e9-8f5d-4a5a3c242346.png)

**z14 Everest aretes** - improved symmetrical pattern - mid width
![z14-everest-aretes-smoother-symmetrical](https://user-images.githubusercontent.com/42757252/56787572-ee2e1c00-6837-11e9-9792-0e06edb5ce0a.png)

**z15 Everest aretes** - improved symmetrical pattern - wide
![z15-aretes-northeast-everest-smoother](https://user-images.githubusercontent.com/42757252/56787592-fd14ce80-6837-11e9-94f2-c11cd574c33c.png)

**z15 Northeast ridge Everest**  - improved symmetrical pattern - wide
![z15-aretes-northeast-everest-smoother](https://user-images.githubusercontent.com/42757252/56787798-9a700280-6838-11e9-9e19-31cea745574e.png)

**z16 Everest aretes** - improved symmetrical pattern - wide
![z16-everest-aretes-smoother](https://user-images.githubusercontent.com/42757252/56787586-f8501a80-6837-11e9-844c-e56ae870c6dc.png)

jeisenbe commented:
## 6) Improved symmetrical arete pattern
- New SVGs `arete-wide.svg` for >=z15 and `arete-mid.svg` for z13-z14
- This commit uses the same ridge patterns as 3) and 5) above: narrow at z13-z14, wide at >=z15

**z13 Everest aretes** - improved symmetrical pattern - mid width
![z13-everest-aretes-smoother-symmetrical](https://user-images.githubusercontent.com/42757252/56787743-6b599100-6838-11e9-8f5d-4a5a3c242346.png)

**z14 Everest aretes** - improved symmetrical pattern - mid width
![z14-everest-aretes-smoother-symmetrical](https://user-images.githubusercontent.com/42757252/56787572-ee2e1c00-6837-11e9-9792-0e06edb5ce0a.png)

**z15 Everest aretes** - improved symmetrical pattern - wide
![z15-aretes-northeast-everest-smoother](https://user-images.githubusercontent.com/42757252/56787592-fd14ce80-6837-11e9-94f2-c11cd574c33c.png)

**z15 Northeast ridge Everest**  - improved symmetrical pattern - wide
![z15-aretes-northeast-everest-smoother](https://user-images.githubusercontent.com/42757252/56787798-9a700280-6838-11e9-9e19-31cea745574e.png)

**z16 Everest aretes** - improved symmetrical pattern - wide
![z16-everest-aretes-smoother](https://user-images.githubusercontent.com/42757252/56787586-f8501a80-6837-11e9-844c-e56ae870c6dc.png)

Adamant36 commented:
Interesting..Very, very, interesting..

jeisenbe commented:
Here are some tests in other areas with the latest commit:

## Cerro de los Huertos - Spain
- https://www.openstreetmap.org/#map=13/40.2864/-5.3290
- rock, scrub
z13 ridges and aretes
![z13-spain-ridges-aretes-smooth](https://user-images.githubusercontent.com/42757252/56789725-8c70b080-683d-11e9-9130-5fce7fac6252.png)

z15 ridges on rock
![z15-cerro-de-los-huertos-ridges](https://user-images.githubusercontent.com/42757252/56789800-c641b700-683d-11e9-8a28-63673816e0da.png)

## Cabeza del Estillo - Spain
- https://www.openstreetmap.org/#map=14/41.0434781/-3.4423236
- Scrub, forest and rock - difficult areas

z13 aretes - thin
![z13-cabeza-del-viejo-aretes-forest-improved](https://user-images.githubusercontent.com/42757252/56789489-f3419a00-683c-11e9-89f5-7c16f5633f19.png)

z15 arete - wide
![z15-cabeza-del-estillo-forest-arete-improved](https://user-images.githubusercontent.com/42757252/56789699-79f67700-683d-11e9-93da-1b4ca6b33b28.png)

## Pasir Putih, Baliem Valley, Indonesia
- https://www.openstreetmap.org/#map=14/-4.04133/138.9413
- limestone ridges with areas of rock, sand and grass (partially mapped)

z14 ridges - thin lines
![z14-pasir-putih-improved](https://user-images.githubusercontent.com/42757252/56791017-03f40f00-6841-11e9-9973-461b70d257a6.png)

z15 ridge - wide line + name label
![z15-pasir-putih-ridge](https://user-images.githubusercontent.com/42757252/56790782-6c8ebc00-6840-11e9-9089-249b25cedd15.png)

imagico commented:
A number of comments here on the alternative-colors implementation:

* credit for the fundamental idea goes to @Ircama who in #2138 first explored this design approach.
* the implementation in the ac-style does not necessarily mean i would support this here in the same form.  This is a test how it works that does not yet fully consider all the implications.
* My choice of not rendering names was deliberate.  Only a relatively small fraction of ridge/arete features have names tagged and of those a fair number have descriptions as names (like _Northeast Ridge_) or names duplicated from peaks.
* Related to that use of natural=ridge is fairly non-uniform world wide and is often widely applied to features that are not actually ridges, mostly labeling geometries for mountain ranges or similar concepts.  It might be that such rendering would counteract such misuse but that is not guaranteed.
* My rendering was mostly oriented at the mapping of ridges in Russia and other former Soviet Union countries which is both fairly extensive and fairly consistent.  The results with such mapping are essentially a variant of what is known as skeletal line relief rendering (German: Gerippelinien) which is fairly common for example in small scale sketch maps with a topography focus (like [here](https://www.researchgate.net/figure/b-Skeletal-lines-Courtesy-Imhof-11-Imhof-E-Cartographic-Relief_fig8_233390625)).
* The patterns are generated with a fairly ad-hoc modification of jsdotpattern which i have not published (the usual with some hardcoded parameters in the code and some non-intuitive parameters to adjust).
* The problem with the directed pattern for arete is obvious.  Aretes are not currently mapped with consistent orientation although this would of course be fairly useful for data users.  
* The biggest problem with implementing this kind of rendering in good quality is the lack of support in Mapnik to define line caps for pattern lines.  This necessarily makes the whole thing somewhat crude.
* This kind of rendering of course aggravates the problems with color mixing of the purple boundaries.

jeisenbe commented:
Thanks for the work you did on this @Ircama! I would be happy to make any changes or improvements that you might suggest. 

> Not rendering names was deliberate. Only a relatively small fraction of ridge/arete features have names tagged and of those a fair number have descriptions as names (like Northeast Ridge) or names duplicated from peaks.

Out of 133 `natural=ridge` that I happened to download for testing (above, mainly Spain, Portugal and France), 11 have names. This is consistent with [taginfo: 12.7%](https://taginfo.openstreetmap.org/tags/?key=natural&value=ridge#combinations) of all features tagged `natural=ridge` are combined with a `name=*`. 

The ridge names that I happened to download appear to be real:
1. "Crête de Coum Grane"
2. "Crête du Liou" (x2)
3. "Crête du Lys"
4. "Crête du Moun Né" 
5. "Cuchillar de las Navajas"
6. "Cuerda de los Pelillos"
7. "Cuerda del Berrueco"
8. "Cuerda del Caramito"
9. "Los Tres Hermanitos"
10. "Riscos de la Portilla Honda"

Taginfo doesn't have combinations for `natural=arete` but [overpass-turbo shows 198](https://overpass-turbo.eu/?Q=%5Bout%3Ajson%5D%5Btimeout%3A120%5D%3B%0A(%0A%20%20way%5B%22natural%22%3D%22arete%22%5D%5B%22name%22%5D%3B%0A)%3B%0A%0Aout%20count%3B&C=-8.05923;19.6875;1) `natural=arete` with `name=*` out of 2026 ways - that's almost 10%

By comparison, currently `natural=cliff` names are rendered, but only 6.4% of `natural=cliff` have a `name=*` tag: https://taginfo.openstreetmap.org/tags/?key=natural&value=ridge#combinations

And `man_made=embankment` names are rendered; only [2.6% have a name](https://taginfo.openstreetmap.org/tags/?key=man_made&value=embankment#combinations)

While it's true that sometimes a ridge and a peak share a name, in my experience there are more peaks incorrectly tagged with a name that belongs to it's ridge than the other way around. 

For example, in my home town area in California, many of the ridge names were imported as peaks from GNIS. It's easy to see the mistakes since many have names like "X Ridge". 

And I've been known to tag a peak with the name of a ridge even here in Indonesia (see Pasir Putih ridge and peak above. It's really the name of the ridge; the high point is not visually prominent)

jeisenbe commented:
> use of natural=ridge is fairly non-uniform world wide and is often widely applied to features that are not actually ridges, mostly labeling geometries for mountain ranges or similar concepts.

I actually haven't seen this yet. Do you know of well-mapped places where this is common?

The main problem I've seen is that most ridges are mapped rather roughly, in the general direction of the ridge but not exactly following the ridgeline. Perhaps this is due to lack of good aerial imagery, but it may be that many people think of these as labeling lines, rather than an exact representation of the ridgeline. Rendering a linear feature will help change this perception.

> The problem with the directed pattern for arete is obvious. Aretes are not currently mapped with consistent orientation although this would of course be fairly useful for data users.

For a while the natural=ridge page said that ridges had to be mapped in an uphill direction, so I was doing this, but it was quite tedious to change the way direction every time the ridge went up and down. Now I found out that this was a rogue wiki edit, and most ridges are not mapped in this way.

This is why I've tried to make non-oriented patterns for aretes. This has taken up the most time so far. 

If we can't get a good arete pattern to work we could just use the same pattern for ridges and aretes. There are only 2000 aretes mapped, and they are a type of ridge. But I hope the new pattern might work. 

> This kind of rendering of course aggravates the problems with color mixing of the purple boundaries.

I'll work on that again, as soon as we deal with the current controversial PR.

jeisenbe commented:
Here's some test renderings from the previously suggested area in the Caucasus, with the current commit.
https://www.openstreetmap.org/#map=13/43.2551/42.8607

z13 ridges and aretes
![z13-south-potop-ridges-aretes-thin](https://user-images.githubusercontent.com/42757252/56812698-5d2a6580-6876-11e9-8f2c-d1367c86dffb.png)

z14 ridges and aretes
![z14-Worehuykoba-aretes-ridges-scree](https://user-images.githubusercontent.com/42757252/56812719-661b3700-6876-11e9-9188-5d65c1987779.png)

z14 ridges
![z14-potop-ridges](https://user-images.githubusercontent.com/42757252/56812738-73382600-6876-11e9-87dc-ac58f3f3de6b.png)

z15 ridges and aretes
![z15-worehujkoba-aretes-ridges](https://user-images.githubusercontent.com/42757252/56812774-84813280-6876-11e9-9463-519adfcac28a.png)

z15 ridges
![z15-potop-ridges-wide](https://user-images.githubusercontent.com/42757252/56812795-8e0a9a80-6876-11e9-9260-bf073c4263a5.png)

The bare_rock pattern is challenging. I wonder if the bare rock pattern really needs to be so much stronger than scree/shingle?

imagico commented:
I did not mean to say that names should for sure not be rendered.  However if mapping ridges is meant to document a physical geography feature (which is the only sensible idea for that tag IMO) matching names to this is often counterproductive because it leads people to delineate abstract named features and not locally verifiable physical structures  And from what i saw looking at the data named ridges are often cases of poorer mapping than unnamed ridges.

> I actually haven't seen this yet. Do you know of well-mapped places where this is common?

Well mapped places? - i am not sure what qualifies as such.  A few examples for uses of ridges for things that are not well defined locally verifiable physical features:

https://www.openstreetmap.org/way/143571111
https://www.openstreetmap.org/way/655973348
https://www.openstreetmap.org/way/530704973
https://www.openstreetmap.org/way/338467628
https://www.openstreetmap.org/way/549656211

Based on this kind of mapping i would generally consider natural=ridge not suitable for rendering here but since there is as explained a considerable volume of consistent use in some parts of the world i would not rule it out if we can make sure that the style of rendering supports this use and discourages other applications.

jeisenbe commented:
> https://www.openstreetmap.org/way/143571111

Cordillera Cutucú Oriental, mapped in 2012 in Ecuador. 
(Spanish-speaking countries use similar words for "ridges" and "(Mountain) ranges": this is a small mountain range). 
This way looks like it consists of 3 ridges, in the north, central and south parts, but they've been connected across a couple of flatish areas, and the southernmost end is iffy (though I am just basing these opinions on the contour lines). It's not great, but could be fixed with a couple minute's work. 

> https://www.openstreetmap.org/way/655973348

Serra da Chibata, Brazil
- Serra means "ridge" or "(mountain) range" in Brazil
- Probably imported: source IBGE by user "importES"
- The southern 30 km  of this way seem to follow the top of a long ridgeline. Then it goes northeast along another ridge, and then north again on a third ridge. The northern third has a couple of flat spots that are mistakes. But overall it looks like most of it follows ridges - it just needs to be broken into smaller ways. 

> https://www.openstreetmap.org/way/530704973

Serra Mundo Novo, Brazil
- This way is drawn rather coarse, perhaps because it's from "IBGE" as well? It should be 2 separate ridges, broken right about the middle (where the two ridges are connected by a small saddle or pass). But I've seen many rivers and coastlines in the tropics that are mapped more coarsely: rendering them helps remind mappers to fix them. 

> https://www.openstreetmap.org/way/549656211

Unnamed ridge in western USA.
The changeset says ""boundary=administrative" is not to be used specifically on ways used to separate areas of woodland"
It looks like these 2 ridges got attached to the natural=wood area and an administrative boundary; perhaps that's why they are joined across a relatively flat area in the middle.

In contrast, I picked a couple of random areas in California, Spain, Chile and Argentina. There were a couple of long ridges in Argentina that need to be broken into smaller pieces, and some of the ridges in California are roughly traced and need to be more precise. But most of the ridges in the places that I checked were quite good, and rendering will be helpful, eg:

South America
https://www.openstreetmap.org/way/462657726
- and the other nearby ridges are caldera rims - much better tagging than natural=cliff

https://www.openstreetmap.org/way/449488940 and https://www.openstreetmap.org/way/449488929
- Many small branching ridges are nicely drawn off of these

https://www.openstreetmap.org/way/449889553 - and around

North America
https://www.openstreetmap.org/way/182814444 - lots of named ridges in the SW USA
https://www.openstreetmap.org/way/452735804

Europe - Spain and France:
- Crête du Moun Né - https://www.openstreetmap.org/#map=14/42.8935/-0.1673
- Cuerda de los Copetes - https://www.openstreetmap.org/#map=13/40.2864/-5.3290
- Cabeza del Viejo - https://www.openstreetmap.org/#map=14/41.0434781/-3.4423236

Nepal:
https://www.openstreetmap.org/#map=13/27.8088/86.5408

I haven't looked in the Alps or Norway, probably too much data to download, but I suspect there are many good examples there?

imagico commented:
Note my critique of the linked to samples is not that they are mapped imprecisely but that they don't represent something that is defined in a verifiable way.  This is a very similar to the problem of place=square - which as a tag is used in a self referential fashion for things called a square (or what a mapper might consider an equivalent to that in some other language) but without there being a way to independently determine if something is a square or not.

Such use of natural=ridge is not the dominating use but it is not a rare one either so i am not sure we can actually render ridges in a way that provides productive mapping support.

jeisenbe commented:
> Note my critique of the linked to samples is not that they are mapped imprecisely but that they don't represent something that is defined in a verifiable way.

Could you clarify this? 

What sort of research can I do that will show if these features are tagged consistently enough for rendering? There are over 60,000 ridges, but I could download a sample from several countries. The thing is, I don't know exactly what problems I'm looking for. 

I don't think you are saying that ridges are not verifiable features. Back in [2014](https://github.com/gravitystorm/openstreetmap-carto/issues/1148#issuecomment-66204385) and [2015](https://github.com/gravitystorm/openstreetmap-carto/issues/1148#issuecomment-68638208) you were in favor of rendering ridges. 

The topographical definition of ridges is quite precise

1) My first guess it that you object to mappers using this tag for mountain ranges which are not one linear ridge. This seems to happen in language areas where the difference between "range" and "ridge" is not so clear (eg "cordillera", "sierra"/"serra" etc). 

A Brazilian mapper mentioned the issue in 2015: https://github.com/gravitystorm/openstreetmap-carto/pull/2138#issuecomment-68698165

_"When I started to map the Serras in this region my understanding was that "Serra" translates best as "ridge". The official IBGE maps which we are allowed to use for OSM simply show in capital letters the approximate location of these Serras, they do not go into any geological details. At the same time, the only documented/approved tag I found on OSM's wiki was natural=ridge. So it was a natural step to map all the Serras as ridges. It is now clear to me that it best to revise this."_

I can check the Portuguese and Spanish translations of the wiki for `natural=ridge` - but I suspect that rendering these features as a continuous line will help make it clear that they are not just name label geometries. 

2) Perhaps also there are too many ridges where one line has been used to map a very long feature which would be better broken up into shorter ways?

This would be analogous to the situation with rivers, where best practice is to keep the lines shorter than 10km, and the water polygons should be even shorter to make editing easier. (Our current way_pixel based filtering at <z10 provides bad incentives about this)

However, this is quite easy to fix by breaking up the long ways. They can even have the same name if the local name is still correct for each ridge segment, or the name may change (just like it does on the Rhine at language borders). 

3) I don't really understand this example: "This is a very similar to the problem of place=square - which as a tag is used in a self referential fashion for things called a square"

It looks like the problem with `place=square` is that it is used for 3 different things, which can all be tagged in another way.

- I. A small urban park in the center of a town, village or city neighborhood, usually rectangular and surrounded by streets (or pedestrian streets) - very common in Latin America.
- II. A paved open area in the middle of a town, village or quarter, traditionally used for parades and markets, also good for demonstrations, protests, etc.
- III. A neighborhood surrounding a square, plaza or traffic circle, which is named for the central square. Eg Harvard Square (Cambridge, Boston), Trafalgar Square (London)

I. Can be mapped as `leisure=park` instead, 
II. Can be mapped as `highway=pedestrian` with `area=yes` and `surface=paved/cobblestone/etc`, III. can be mapped as `place=neighborhood` (and is mistagged if mapped as place=square according to the wiki definition)
So `place=square` this is an ambiguous tag but there are better alternatives (all of which have a name label rendering in this style).

I don't see this problem with `natural=ridge`: there are no correct alternative tags. The tag `natural=peak` is clearly defined as a high point, not a linear feature. 

The one issue is mistagging of mountain ranges as ridges. These should probably be tagged as `natural=mountain_range` - however that tag is not very widely used and not well documented, so it's not surprising that `natural=ridge` is sometimes misused in this way.

imagico commented:
> What sort of research can I do that will show if these features are tagged consistently enough for rendering? There are over 60,000 ridges, but I could download a sample from several countries. The thing is, I don't know exactly what problems I'm looking for.

I don't think there is anything you could do from the mapping side.  As you said there are tens of thousands of features and a significant fraction of them is with a strong variability in what exactly they are applied to (as illustrated with examples).  It would of course be interesting to analyze more precisely what fraction this is and what classes of use can be identified.

> I don't think you are saying that ridges are not verifiable features. Back in 2014 and 2015 you were in favor of rendering ridges.

In the course of implementing the ac-style rendering i looked at the data more closely than before and this changed by position a bit.

I will try to explain it from a slightly different perspective: How would you define a natural=ridge tag from scratch if you wanted it to be clear and locally verifiable?  That would need to be based on the local relief.  And in many, especially in relatively flat areas or with small scale erosion, you would need to define some quantitative cutoff - otherwise you could end up with an almost arbitrarily dense set of ridges that can be mapped in most areas.

The current wiki description provides no really meaningful definition.  The _continuous elevated crest_ (which i added back in 2014, which however turned out to be quite insufficient as a discerning definition) is functionally equivalent to defining a ridge as any watershed divide.  Based on this actual use in the database as we discussed like the labeling geometries or the >100km long lines drawn along some watershed considered to be the main crest of some mountain range and labeled accordingly are often not formally wrong according to the definition.  Yet for the data users they are something completely different than what you for example see in the Caucasus and elsewhere.

A clear new definition for a ridge tag would be _a line of continuous negative earth surface curvature with a continuous curvature radius of less than X meters_ - with X being something like 20-50m probably.  The problem with that is that it would be interrupted at every saddle point (which by definition has zero curvature) - so you might want to introduce some skipping rule for small saddles - which however could affect local verifiability.  Note use of natural=arete probably follows such a definition more closely already with a much smaller value for X.

Anyway - this is an abstract exercise, we are not inventing a new tag here but dealing with an existing one in wide use.  I guess my question for the decision on rendering is if we can have the reasonable expectation that use of natural=ridge will develop into something that follows a uniform, locally verifiable definition (if similar to the one i made up above or a different one does not really matter) and if we can with a suitable rendering support such a development.  This is what i am looking for here.

Because if use of natural=ridge stays as broad and undefined in meaning as it is right now the more responsible decision would be to recommend to mapper to create and use a new, better defined tag instead for what they want to map (as it happened in other cases in the past - like leaf_cycle/leaf_type instead of wood=*).

verdy-p commented:
Aretes and ridges are *directional* objets: if you use chevrons to render them, they should be oriented consistantly, depending on the altitude, just like river ways, and should not depend directly on how the way was drawn (where the altitudes are kwown between nodes, and there should be a node at each summit/peak and at each minimum).

You may also consider the difference of altitude between marked nodes to compute the way length and the average steep: this can be used to customize the length of the chevron strokes (or its apparence like angle relative to the main way of the arete or ridge): its not uncommon for ridges and aretes to have "flat" sections (no significant change of altitude along them, even if the half planes of on the two sides are sharply angled).

Finally ridges and aretes should have a way to define how sharp they are (i.e. the relative angle of the two orthogonal lines on each side)

This also applies to cliffs (mostly horizontal on one side, but nearly vertical on the other side). This can generalized (cliffs, ridges, aretes) by tagging the steep angles for orthogonal directions to the left and right of the way (these two angles are not necessarily the same, especially for cliffs, and for most rocky or icy aretes).

This could as well apply to some large buildings or infrastructures (e.g. dams), or even in large enough pedestrian areas (transition betwen a flat surface and a steeply inclined surface, when there surfaces are significantly larger than a street, or transition by very wide stairs not well repredented as ways in OSM, but where drawing each "step" is not very useful of not suitable beause there are no steps but only an softly inclined plane, for example in plazzas around large elevated monuments or buildings).

The idea is then to tag elevations on nodes of the geometry of ways, plus additional tags for angles on the left and right side of ways (but only if this is a linear feature).

For area features, we don't need these additional angle tags, we just need altitudes on at least 3 nodes of the closed polygon, it is eanough to compute the angles by splitting the polygon into triangular facets that are the closest to equilateral, i.e. whose the sharpest of their 3 angles are maximized: there are good triangulation algorithms used in CAD for 3D modeling of a surface sampled on a more or less regular grid of nodes).

And for these computed facets, the ridge/edge/cliffs can be determined automatically without even drawing them as ways in OSM (it is already useful for numeric terrain models, already rendered in some OSM maps: the isobaths are computed and interpolated directly from a sampled grid of elevation measurements and this requires much less data than drawing all ways for each altitude or depth by step of 10 meters: the grid of elevations is triangularized, then the skeleton of riges/areates are determined, then nodes at each elevation step along this skeleton are computed, then nodes are connected between each side of the triangular skeleton, finally these generated polylines are smoothed by quadratic or cubic Beziers and altitude values can be texted along these Beziers)

jeisenbe commented:
@imagico - I take your comments above (https://github.com/gravitystorm/openstreetmap-carto/pull/3767#issuecomment-487270565) as rejecting rendering of `natural=ridge` due to inconsistent use of the tag on both sharply delineated ridges, as well as on drainage divides where there is not a sharp ridgeline but a rounded, gradual slope. Is that correct?

That would suggest using a different style for `natural=arete` (more similar to a two-sided `natural=cliff`), in a different PR.

imagico commented:
No, i am not rejecting idea, i am just suggesting to critically evaluate if the proposed rendering (in terms of design and starting zoom levels) provides constructive mapper feedback in support of consistent use of the tag.

The dominant use of natural=ridge is clearly for mapping skeleton lines in (mostly glacially shaped) mountain areas, as it is widely done in Russia and former Soviet countries.  Rendering should ideally positively support such use and discourage use on other things like for mapping mountain ranges with labeling lines etc.

jeisenbe commented:
> Rendering should ... discourage use on other things like for mapping mountain ranges with labeling lines etc.

Could this be done by avoiding rendering at mid-zoom levels, starting at z14 only? This would help suggest that the rendering is not for large mountain ranges. 

We could also avoid rendering names for now, since `natural=mountain_range` is not rendered, so that mappers are not tempted to use `natural=ridge` as way to get mountain ranges rendered.

imagico commented:
Yes, not starting rendering too early and not rendering names (or rendering them later/less prominently) could definitely help.  As said i am fairly unsure about this - would need to test the particular design suggestion and would probably also need to evaluate the effect carefully after roll-out.

pnorman commented:
This PR implements 6) Improved symmetrical arete pattern, right?

I'm broadly-speaking in favour, but don't feel qualified to review it since there aren't natural=ridge features where I normally map.

jeisenbe commented:
The latest test images in https://github.com/gravitystorm/openstreetmap-carto/pull/3767#issuecomment-486959048 show the current commit, with "6) Improved symmetrical arete pattern" for aretes, and the same ridge pattern as 3) "Narrower ridges at z13 and z14".

pnorman commented:
Can we take the [WIP] label off of this PR and come to a decision about it, or is more work still needed?

jeisenbe commented:
I've updated the PR by changing the initial zoom level to z14 for the line rendering of aretes and ridges, and I've renamed the svg files while deleting those that are no longer needed.

For now I've left the name rendering at z15, pending review.

jeisenbe commented:
Rebased for v5.0.0, but I think we need input and perhaps a review from @imagico. 

Since there are concerns about how natural=ridge is being used around the world, perhaps someone who has a global server would be willing to test this PR, so we can check a larger variety of areas?

jeisenbe commented:
Ok, some areas checked (these are pbf extracts which I've already downloaded, picked somewhat randomly, the data may be 3 to 12 months old). Probably helpful to view these with the opencyclemap layer to see terrain.

### Karlovarsky, Poland (28 mb file)
1 `natural=ridge` ([way 544249714](https://www.openstreetmap.org/#map=17/50.34703/12.93766)) - unnamed, short, in woods near some cliffs, looks ok.
0 `natural=arete`

### Armenia (25 mb file)
221 `natural=ridge`:
- 3 with geological=*: e.g. [way 520468701](https://www.openstreetmap.org/way/520468701) - caldera_rim, circular, unnamed - ok
- 1 shared with highway=path - [way 510258661](https://www.openstreetmap.org/way/510258661) - leads to peak on a volcanic crater/caldera
- 1 shared with boundary=protected_area - [way 554230518](https://www.openstreetmap.org/way/554230518) - connects to a natural=cliff at the west side:

<img width="600" alt="ridge-way-554230518" src="https://user-images.githubusercontent.com/42757252/76190864-5a7ff700-6221-11ea-834f-f0e752dcf1c4.png">


The rest are unnamed, no other tags. Random examples:
- [way 510258644](https://www.openstreetmap.org/way/510258644) - ridge ending at a peak on one end and a saddle on the other
- [way 552030717](https://www.openstreetmap.org/way/552030717) - connects 4 peaks and 1 saddle

<img width="611" alt="ridge-way-552030717" src="https://user-images.githubusercontent.com/42757252/76174625-d1020200-61eb-11ea-9934-70c6ff8b20e1.png">

- [way 309355567](https://www.openstreetmap.org/way/309355567) - 2 other ridges across river
- [way 739533281](https://www.openstreetmap.org/way/739533281) - ends near communication towers
- [way 554722577](https://www.openstreetmap.org/way/554722577) - cliffs at east end:

<img width="584" alt="ridge-way-554722577" src="https://user-images.githubusercontent.com/42757252/76190828-43410980-6221-11ea-93f6-3c5a111d40a5.png">

Other areas with more landcover and cliffs mapped:

<img width="584" alt="ridge-armenia-cliffs-15:39 6195:45 6029" src="https://user-images.githubusercontent.com/42757252/76179415-2fd07700-61fe-11ea-90b9-c73e455681a3.png">

<img width="584" alt="ridge-armenia-14:39 6602:45 6494" src="https://user-images.githubusercontent.com/42757252/76179418-3232d100-61fe-11ea-9968-5d4d98743472.png">


## Wales, UK: (73 mb)
- 14 natural=ridge ways
- 1 natural=arete ways

Ridges: 5 named, 9 unnamed.

<img width="599" alt="ridge-wales-heath-ogof-ddu-14:52 7771:-3 6551" src="https://user-images.githubusercontent.com/42757252/76179434-3bbc3900-61fe-11ea-873e-f178d4e598f4.png">

Named:
[way 37917539](https://www.openstreetmap.org/way/37917539), "name": "Ogof Ddu", - oops, crosses a stream (above)
[way 223985600](https://www.openstreetmap.org/way/223985600),  "ele": "547", "name": "Carreg Blaen-Llym":

<img width="596" alt="ridge-Carreg-Blaen-Llym-15:52 9768:-3 9897" src="https://user-images.githubusercontent.com/42757252/76175450-01e43600-61f0-11ea-8cf2-702f890968d1.png">

[way 276120618](https://www.openstreetmap.org/way/276120618), "name": "Y Gribin",
[way 334433650](https://www.openstreetmap.org/way/334433650), "highway": "path", "name": "Bristly Ridge":

<img width="567" alt="ridge-bristly-wales-16:53 1078:-4 0026" src="https://user-images.githubusercontent.com/42757252/76175213-c9902800-61ee-11ea-9fc5-353c6feaacdf.png">

[way 755441509](https://www.openstreetmap.org/way/755441509), "name": "Pant-y-gwair",

<img width="583" alt="ridge-crib-goch-wales" src="https://user-images.githubusercontent.com/42757252/76175605-c138ec80-61f0-11ea-81e1-734b87a5a340.png">

Unnamed (samples: first, middle and last):
[way 24437734](https://www.openstreetmap.org/way/24437734) - (above) next to Crib Goch peak, crib Goch path, and some cliffs. Also another unnamed ridge nearby connects.
[way 755441509](https://www.openstreetmap.org/way/755441509) - rocky ridge at end of cape in sea, with small peaks and saddles mapped: https://www.openstreetmap.org/#map=18/51.5699930/-4.1108005

<img width="453" alt="ridge-cape-wales-z18" src="https://user-images.githubusercontent.com/42757252/76179392-1b8c7a00-61fe-11ea-826c-2751e5f414bd.png">

[way 24471527](https://www.openstreetmap.org/way/24471527) - ridge in health, around end of stream: 

<img width="520" alt="ridge-wales-way-24471527-14:52 6110:-3 9373" src="https://user-images.githubusercontent.com/42757252/76175416-d8c3a580-61ef-11ea-874c-bc89885018dd.png">


## Shikoku, Japan (67.5 mb)
- only 3 natural=ridge ways
- 0 natural=arete ways

Ridges:
- [way 194043690](https://www.openstreetmap.org/way/194043690) and way 194044504, "natural": "ridge", "source": "YahooJapan/ALPSMAP", "yh:WIDTH": "1.5m未満" - this one comes down all the way to the river in the NE:

<img width="498" alt="ridge-japan-2ways-194043690-16:33 2703:132 5379" src="https://user-images.githubusercontent.com/42757252/76179387-15969900-61fe-11ea-9654-bb18acb1819b.png">

-[way 453108986](https://www.openstreetmap.org/way/453108986) - no other tags. Seems right:

<img width="600" alt="ridge-japan-way-453108986-15:33 0338:132 4979" src="https://user-images.githubusercontent.com/42757252/76175788-82effd00-61f1-11ea-903b-96bfd370d469.png">


## Latvia (65 mb)

6 natural=ridge ways, no aretes. None have a name or any other tags.
- way 723177311, way 723179931, way 723182377, way 723182378: this is like a archeological site? - https://www.openstreetmap.org/#map=16/56.9674/23.1326

<img width="600" alt="latvia-ridges-16:56 9674:23 1326" src="https://user-images.githubusercontent.com/42757252/76176432-fb57bd80-61f3-11ea-9e11-a647d3477424.png">

- way 723184524 - along a cliff, one or the other is incorrect (probably the natural=cliff?)

<img width="594" alt="latvia-ridge-cliff-way-16:56 9715:23 1758" src="https://user-images.githubusercontent.com/42757252/76176431-f8f56380-61f3-11ea-855e-9e91c52815c1.png">

- way 765999687 - in an old fort, by a "star fort" with lots of walls, embankments and ditches. 


## Cape Verde Islands (7 mb)
- 5 natural=ridge ways, all together:
https://www.openstreetmap.org/#map=15/16.8094/-24.9415
<img width="532" alt="cape-verde-ridges-15:16 8094:-24 9415" src="https://user-images.githubusercontent.com/42757252/76179203-5e018700-61fd-11ea-8d0f-b5b475198b4c.png">


## Costa Rica (22 mb)
- 7 natural=ridge ways, all added in 1 changeset, no other tags (though a couple share nodes with administrative boundaries):
Way: 560663214, Way: 560663215, Way: 560663216, Way: 560663219, Way: 560663243, Way: 560682809, Way: 560732278

Example: way 560663243
<img width="532" alt="costa-rica-ridge-14:10 2382:-85 3139" src="https://user-images.githubusercontent.com/42757252/76179287-ba64a680-61fd-11ea-84ed-73ee6ab96fe9.png">


## Sogn Og Fjordane region, Norway (29 mb):
1 ridge only (surprisingly): - named (same as peak). No aretes.
[way 429492690](https://www.openstreetmap.org/way/429492690) - 
https://www.openstreetmap.org/#map=15/61.0534/5.8184

<img width="520" alt="norway-named-ridge-14" src="https://user-images.githubusercontent.com/42757252/76179681-0fed8300-61ff-11ea-8b67-ce9f0a919094.png">

<img width="609" alt="norway-named-ridge-15:61 0534:5 8184" src="https://user-images.githubusercontent.com/42757252/76179680-0bc16580-61ff-11ea-961a-6cb320b56965.png">


## Corsica, France (25.6 mb)
11 natural=ridge, all named. 1st, middle and last:
1) Way: a fursiccia (662696095)
https://www.openstreetmap.org/#map=15/42.1409206/8.8684606

<img width="601" alt="corsica-ridge-a-fursiccia-15:42 1401:8 8665" src="https://user-images.githubusercontent.com/42757252/76180000-0d3f5d80-6200-11ea-98b1-0762f04213a7.png">

2) Way: Crete du Spugnolu (369084556)
https://www.openstreetmap.org/#map=15/42.1185889/8.7165095

<img width="601" alt="corsica-ridge-crete-15:42 1185889:8 7165095" src="https://user-images.githubusercontent.com/42757252/76180064-3f50bf80-6200-11ea-8a9d-e960afeba925.png">

3) Way: u suale (662696099)
https://www.openstreetmap.org/#map=15/42.1768808/8.7362089

<img width="559" alt="corsica-ridge-u-suale-14:42 1769:8 7362" src="https://user-images.githubusercontent.com/42757252/76180167-95bdfe00-6200-11ea-8bb9-307d3d0021b2.png">


## Hautes Alpes, France
22 aretes, 2 with names (finally):

Way: Pointes de la Condamine (359540655)
https://www.openstreetmap.org/#map=16/44.8522/6.3496
<img width="505" alt="france-pointes-arete-16:44 8522:6 3496" src="https://user-images.githubusercontent.com/42757252/76180687-00236e00-6202-11ea-9915-dae3db8c099c.png">

Way: Rochers de la miglia (430618548)
https://www.openstreetmap.org/#map=15/45.0660081/6.597636

<img width="505" alt="arete-15:45 0660:6 5976" src="https://user-images.githubusercontent.com/42757252/76189471-2525da00-621e-11ea-9abc-d5a97240d7f2.png">

86 natural=ridge, 45 with names:
1st, middle and last example:

1) https://www.openstreetmap.org/#map=15/44.7394/6.3419
- way: 181468331    "boundary"="administrative" "source"="Bing" "natural"="ridge" "admin_level"="8"
<img width="505" alt="hautes-alpes-ridge1-15:44 7394:6 3419" src="https://user-images.githubusercontent.com/42757252/76189480-2a832480-621e-11ea-98a5-46af532dae2a.png">

2) https://www.openstreetmap.org/#map=14/44.6673/6.4349
Way: 514891635 "natural=ridge"
<img width="562" alt="hautes-alpes-ridge2-14:44 6673:6 4349" src="https://user-images.githubusercontent.com/42757252/76189709-b72de280-621e-11ea-942f-e0a8700f4065.png">

3) https://www.openstreetmap.org/#map=16/44.5012962/5.7956054
Way: Serre Mourresque (435014767) "natural=ridge" name="Serre Mourresque"
<img width="562" alt="hautes-alpes-ridge3-16:44 6673:6 4349" src="https://user-images.githubusercontent.com/42757252/76190127-a3cf4700-621f-11ea-9131-928e8982afd3.png">

## Honduras:
1 natural=ridge:
Cordillera de Sulaco (396380445) - 170 km long, this is actually a mountain range. As mentioned before, this seems to be a problem in Spanish-speaking countries.

## Extremadura, Spain:
5 natural=arete features, 2 with names
Way: 656626264 - https://www.openstreetmap.org/#map=15/40.255936/-5.3934382
Way: 656626908 - https://www.openstreetmap.org/#map=15/40.2623974/-5.406605
Way: 659659765 - https://www.openstreetmap.org/#map=15/40.2132022/-5.5861352
Way: Cuerda de la Hoya (651645785) - https://www.openstreetmap.org/#map=15/40.2708149/-5.337791
Way: Riscos del Bollo (656626263) - https://www.openstreetmap.org/#map=15/40.2607993/-5.3972255 - shown, also 2 others and a ridge:
![Uploading extremadura-arete-riscos-del-bollo-15:40.2606:-5.4007.png…]()

16 natural=ridge features, 7 with names:
1) Way: 655143233 - https://www.openstreetmap.org/#map=15/40.2578915/-5.3538951
<img width="546" alt="extremadura-ridge1-14:40 2579:-5 3539" src="https://user-images.githubusercontent.com/42757252/76190956-97e48480-6221-11ea-9644-a4088a40b154.png">

2) Way 660103598 - https://www.openstreetmap.org/#map=14/40.2163/-5.4681
<img width="546" alt="extremadura-ridge2-14:40 2163:-5 4681" src="https://user-images.githubusercontent.com/42757252/76192070-3a9e0280-6224-11ea-8eb3-414a2a970179.png">

3) Way: Sierra de Monfragüe (491109988) - https://www.openstreetmap.org/#map=15/39.8253209/-6.0336685
![Uploading extremadura-sierra-3-15:39.8253209:-6.0336685.png…]()


## Bhutan
2 ridges:
Way: Nawaten (690677756) - https://www.openstreetmap.org/#map=15/27.2111/88.9115
<img width="546" alt="bhutan-ridge-1-nawaten" src="https://user-images.githubusercontent.com/42757252/76191857-a5027300-6223-11ea-9b46-38e26ea6d451.png">

Way: 469614592 - https://www.openstreetmap.org/#map=14/27.248865/89.6989217
<img width="546" alt="bhutan-ridge-2" src="https://user-images.githubusercontent.com/42757252/76192049-2eb24080-6224-11ea-83d0-33d497bd6124.png">

## No Ridges or aretes:
* I checked these places but found no features:
### Estonia
### Uruguay
### Singapore:
### Congo (Brazzaville):
### Burundi:

Looking at all of these, I think a large majority of uses of natural=ridge are appropriate and will work with this proposed rendering, and there are enough named ridges for the name label to be reasonable, though there are still a few that need to be better mapped and a few that should be natural=mountain_range. The aretes seem to be fine too.

jeisenbe commented:
>  concerns about the arete pattern - it appears fairly mushy

I was trying to compromise between showing aretes in a similar way to ridges, or making them more like cliffs.

You've approved this PR, so do you think we should go with this for now, or should I try some other options for the aretes first?

(Either way, I will wait till after v5.0.0)

imagico commented:
> You've approved this PR, so do you think we should go with this for now, or should I try some other options for the aretes first?

I think the current design is acceptable - but definitely could be improved.

jeisenbe commented:
Tests with arete_x4.svg design from https://github.com/gravitystorm/openstreetmap-carto/pull/3767#pullrequestreview-375569188 for z15 and above:

 https://www.openstreetmap.org/#map=15/40.2577/-5.3925
![arete2-15:40 2577:-5 3925](https://user-images.githubusercontent.com/42757252/77153644-79ae3c80-6add-11ea-94a4-143504416793.png)

 https://www.openstreetmap.org/#map=15/40.2663/-5.3367
![arete2-15:40 2663:-5 3367](https://user-images.githubusercontent.com/42757252/77153485-2b00a280-6add-11ea-8dc8-6a1479ae7d0c.png)

 https://www.openstreetmap.org/#map=15/40.2119/-5.5792
![arete2-15:40 2119:-5 5792](https://user-images.githubusercontent.com/42757252/77153491-2d62fc80-6add-11ea-9796-3b0ba5ade6ac.png)

 https://www.openstreetmap.org/#map=16/40.2119/-5.5792
![arete2-16:40 2119:-5 5792](https://user-images.githubusercontent.com/42757252/77153498-305ded00-6add-11ea-8d37-24e7ca212f9e.png)

jeisenbe commented:
The z14 rendering is unchanged. I've cleaned up the files to remove unneeded metadata.
